File: src/static/Code/Python/Challenge015.py
import calendar

leapYears=[]
for year in range(1016, 1996+1, 20):
    # 是闰年并且该年1月1日是星期四
    if calendar.isleap(year) and calendar.weekday(year,1,1)==3:
        leapYears.append(year)
print(leapYears[-2])

File: src/static/Code/Go/Challenge015.go
package main

import(
	"fmt"
	"time"
)

func (c *Challenge) Challenge015() {
	var leapYears []int
	for year := 1016; year <= 1996; year+=20 {
    	// 是闰年并且该年1月1日是星期四
		if isLeapYear(year) {
			date := time.Date(year, time.January, 1, 0, 0, 0, 0, time.UTC)
			if date.Weekday()==4 {
				leapYears=append(leapYears, year)
			}
		}
	}
	fmt.Println(leapYears[len(leapYears)-2])
}

func isLeapYear(year int) bool {
	return year%100==0 && year%400==0 || year%4==0
}